### ISSUE TITLE
feat: add multi-reference XTTS voice cloning workflow and recording guide

### ISSUE THREAD

**Issue by onurozkir** [onurozkir/De-ReVot#20]
## Summary
Upgrade the XTTS-v2 voice cloning pipeline in `teams-translator` to natively support multi-reference audio conditioning (specifically a curated set of 6 clean reference WAVs of 8–10 seconds each) instead of relying on a single reference audio file. This feature introduces automated multi-WAV profile discovery, composite deterministic latent caching, strict yet practical audio validation, and comprehensive documentation with recommended Turkish recording scripts.

---

## Motivation
Currently, voice cloning fidelity and cross-lingual pronunciation stability in XTTS-v2 can be uneven when conditioned on a single, short (5–8s) recording or an arbitrary long clip. XTTS-v2's speaker encoder (`speaker_embedding`) and autoregressive prefix conditioner (`gpt_cond_latent`) yield significantly higher speaker similarity, cleaner timbre, and resilient prosody when provided with multiple phonetically and syntactically diverse samples recorded under identical acoustic conditions.

Transitioning to a structured 6-sample dataset (neutral, assertive, narrative/long, question/pauses, technical/named-entity, and Turkish-phoneme/numerical) resolves acoustic under-conditioning while preserving low-latency runtime inference via precomputed conditioning caching.

---

## Current Behavior
1. **Model Initialization**: `XTTSv2Adapter.initialize()` in [`src/teams_translator/tts/xtts_backend.py`](src/teams_translator/tts/xtts_backend.py#L69-L104) loads the model using `Xtts.init_from_config()` and `model.load_checkpoint()`. Instantiated in [`src/teams_translator/main.py`](src/teams_translator/main.py#L42) and initialized via [`src/teams_translator/streaming/orchestrator.py`](src/teams_translator/streaming/orchestrator.py#L88-L100).
2. **Profile Discovery & Data Model**: [`src/teams_translator/tts/base.py`](src/teams_translator/tts/base.py#L12-L25) defines `VoiceProfile` with a single `reference_audio_path: str`.
3. **Manifest Loading**: [`src/teams_translator/tts/conditioning.py`](src/teams_translator/tts/conditioning.py#L16-L70) (`VoiceProfileManager.load_profiles()`) scans `voices/` subdirectories, expects `profile.json`, and extracts only a single `"reference_audio_path"`.
4. **Conditioning Extraction**: [`src/teams_translator/tts/xtts_backend.py`](src/teams_translator/tts/xtts_backend.py#L129-L177) computes SHA-256 over that single audio file and passes `audio_path=[str(ref_path.resolve())]` to `model.get_conditioning_latents()`.
5. **Caching**: In-memory dictionary `self._latents_cache` and disk cache in `voices/<profile>/cache/latents_<profile_id>_<sha256>.pt`.
6. **Validation**: Only checks `ref_path.exists()`. No validation for duration, sample format, RMS silence, or multi-sample consistency.
7. **Documentation**: [`README.md`](README.md) describes placing a single 6–10 second `reference.wav` inside `voices/<profile_name>/`.

---

### Directory Layout
Users can provide a speaker directory under `voices/<speaker_id>/` with either individual WAV files or an optional `profile.json`:

```text
voices/
└── default/
    ├── profile.json           # Optional or auto-generated manifest
    ├── voice_01.wav           # Core voice identity (neutral)
    ├── voice_02.wav           # Identity reinforcement
    ├── voice_03.wav           # Long sentence & natural prosody
    ├── voice_04.wav           # Pauses, short utterances & question intonation
    ├── voice_05.wav           # Technical jargon & proper nouns
    ├── voice_06.wav           # Turkish phonetics, numbers & conversational tone
    └── cache/
        └── latents_<hash>.pt  # Composite SHA-256 precomputed conditioning
```

### Profile Manifest Schema (`profile.json`)
Support both single-file legacy format and new multi-reference list or directory auto-discovery:
```json
{
  "id": "1",
  "display_name": "Default Voice (Multi-Reference)",
  "backend": "xtts_v2",
  "reference_audio_paths": [
    "voice_01.wav",
    "voice_02.wav",
    "voice_03.wav",
    "voice_04.wav",
    "voice_05.wav",
    "voice_06.wav"
  ],
  "reference_language": "tr",
  "target_languages": ["en", "fr"],
  "is_default": true
}
```
If `reference_audio_paths` is omitted, `VoiceProfileManager` automatically discovers and sorts all `voice_*.wav` or `*.wav` files in the profile directory (ignoring files inside `cache/`).

---

## Recommended Recording Set

To ensure optimal voice cloning without model fine-tuning, the 6 reference files must be recorded by the **same speaker**, with the **same microphone**, at the **same distance and room acoustics**, with **no background noise or aggressive post-filters (no noise suppression/OBS filters)**:

| File | Purpose | Target Duration | Recommended Turkish Text |
|---|---|---|---|
| `voice_01.wav` | **Ana Ses Kimliği**: Doğal ve nötr konuşma kimliği | ~8–10s | *"Bugün üzerinde çalıştığım projeyi tamamlamak için birkaç farklı yöntemi karşılaştıracağım. Önce mevcut sonuçlara bakıp daha sonra nasıl devam edeceğime karar vereceğim."* |
| `voice_02.wav` | **Ses Kimliğini Güçlendirme**: Tek örneğe aşırı bağımlılığı (overfitting) kırma | ~8–10s | *"Aslında mesele düşündüğümüz kadar karmaşık değil. Sorunun nereden kaynaklandığını doğru şekilde belirlersek geri kalan kısmını çok daha kolay çözebiliriz."* |
| `voice_03.wav` | **Uzun Cümle ve Doğal Prosody**: Akıcı kelime geçişleri ve ritim | ~8–10s | *"Sistemin düzgün ve kararlı çalışabilmesi için bütün bileşenlerin birbiriyle doğru şekilde iletişim kurması, verilerin zamanında işlenmesi ve sonuçların dikkatlice değerlendirilmesi gerekiyor."* |
| `voice_04.wav` | **Doğal Duraklamalar & Soru Tonlaması**: Kısa ifadeler, soru intonasyonu | ~8–10s | *"Tamam, bunu deneyebiliriz değil mi? Önce bir test yapalım. Sonuç doğru mu? Emin miyiz? Bir problem varsa ayarları değiştirip tekrar deneyebilir miyiz? Bu bizim için ne ifade ediyor?"* |
| `voice_05.wav` | **Özel İsimler & Yoğun Teknik Konuşma**: Teknik artikülasyon ve vurgular | ~8–10s | *"OpenAI, NVIDIA, Microsoft Teams, Python üzerinde çalışan sistemde GPU kullanımı, inference latency, model cache ve speaker embedding değerlerini birlikte kontrol edeceğiz."* |
| `voice_06.wav` | **Türkçe Karakterler & Sayılar**: `ç, ğ, ı, ö, ş, ü`, sayılar ve tarih artikülasyonu | ~9–10s | *"Beşinci ayın ikisinde saat dokuz buçukta görüşeceğiz. Üç farklı çözümü karşılaştırıp yirmi dört dosyayı inceleyeceğiz; işimiz yaklaşık kırk beş dakika sürerse öğleden önce bitirebiliriz ya da bir gün daha bekleriz."* |

---

## Proposed Code Changes

1. **`src/teams_translator/tts/base.py`**:
   - Update `VoiceProfile` dataclass:
     - Change `reference_audio_path: str` to `reference_audio_paths: List[str]`.
     - Provide a `@property reference_audio_path -> str` for backward compatibility returning `self.reference_audio_paths[0]`.
2. **`src/teams_translator/tts/conditioning.py`**:
   - Update `VoiceProfileManager.load_profiles()`:
     - Detect `reference_audio_paths: list[str]` or single `reference_audio_path: str`.
     - Implement auto-discovery for `voice_*.wav` in the directory if no audio path is explicitly provided in manifest.
     - Implement `compute_multi_audio_hash(audio_paths: List[str]) -> str` combining individual file hashes in a deterministic sorted order.
3. **`src/teams_translator/tts/xtts_backend.py`**:
   - Update `prepare_voice_profile(self, profile: VoiceProfile)`:
     - Pass list of resolved path strings `audio_path=[str(p) for p in profile.reference_audio_paths]` to `self.model.get_conditioning_latents()`.
     - Hash key: `f"{profile.id}_{composite_hash}"`.
   - Update `synthesize_committed()`:
     - Use composite cache key for fast zero-latency in-memory dictionary lookup.

---

### Multi-Reference Computation Flow
```text
[ voice_01.wav, ..., voice_06.wav ]
              │
              ▼
   Deterministic Sort & Audio Validation
              │
              ▼
   Check In-Memory Cache (_latents_cache[cache_key])
              │── Hit ──> [ gpt_cond_latent, speaker_embedding ] ──> Zero-latency synthesis
              ▼ Miss
   Check Disk Cache (latents_<profile>_<composite_hash>.pt)
              │── Hit ──> torch.load() ──> Store in memory ──> Synthesis
              ▼ Miss
   XTTS model.get_conditioning_latents(audio_path=[...], gpt_cond_len=30, max_ref_length=60)
              │
              ├── speaker_embeddings: mean-pooled across all 6 audios
              └── gpt_cond_latents: extracted from concatenated full audio
              │
              ▼
   torch.save() to disk cache & cache in memory ──> Fast subsequent streaming synthesis
```
- Multi-sample extraction runs once on first profile load and is cached to disk.
- Runtime per-utterance synthesis maintains **0 ms conditioning latency** because precomputed latents are passed directly into `model.inference()`.

---

## Validation
Implement dedicated validation before computing conditioning latents:
- **Existence**: Check that all specified reference audio files exist on disk.
- **Empty / Corrupted check**: Verify file size > 4 KB and readable by `soundfile`.
- **Duration check**: Warning if an individual clip is `< 3.0s` (insufficient phonemes) or `> 30.0s` (excessive memory; XTTS truncates at `max_ref_length`).
- **RMS Energy check**: Reject near-silent audio (`RMS < 0.005`).
- **Channel / Sampling**: Log warning if sample rates differ, but rely on XTTS's native internal resampling (`load_sr=22050`) to avoid unnecessary lossy pre-conversion.
- **Clear Error Messaging**: Fail early with descriptive exceptions indicating the offending file path.

---

## Backward Compatibility
- If `profile.json` has `"reference_audio_path": "reference.wav"` (single file), the system automatically treats it as `reference_audio_paths = ["reference.wav"]`.
- Existing single-WAV voice profiles (such as `voices/anime-girl/reference.wav`) continue working without modification.
- The `VoiceProfile.reference_audio_path` attribute remains accessible via property getter.

---

## Documentation Changes
1. **`README.md`**:
   - Update `Adding Custom Voice Profiles` section to highlight the recommended 6-WAV recording setup.
   - Show updated directory tree with `voice_01.wav` .. `voice_06.wav`.
2. **`docs/voice-recording-guide.md`** (New File):
   - Detailed recording best practices (room acoustics, mic positioning, gain staging, avoiding noise cancelers).
   - Full Turkish reading scripts for `voice_01.wav` through `voice_06.wav` with guidelines on natural reading pace.

---

## Files Likely Affected
- `src/teams_translator/tts/base.py` (`VoiceProfile` dataclass)
- `src/teams_translator/tts/conditioning.py` (`VoiceProfileManager`, hashing, and discovery)
- `src/teams_translator/tts/xtts_backend.py` (`prepare_voice_profile` multi-file handling and caching)
- `src/teams_translator/config/models.py` (`VoiceConfig` if directory conventions are formalized)
- `config/default.toml`
- `README.md`
- `docs/voice-recording-guide.md` (new documentation)
- `tests/unit/test_adapters.py` & `tests/unit/test_xtts_runtime.py`

---

## Testing
- Unit test: `VoiceProfileManager` loads single-file manifest and multi-file manifest correctly.
- Unit test: `VoiceProfileManager` automatically discovers 6 sorted WAV files from a folder when manifest is absent or minimalist.
- Unit test: Deterministic composite hash generation across multiple audio files.
- Unit test: Validation raises expected exceptions for missing, silent, or corrupted WAV files.
- Integration test: Multi-reference conditioning generates valid `(gpt_cond_latent, speaker_embedding)` tensors and successfully caches to disk.
- Mock test: Pipeline runs end-to-end with mock TTS using multi-reference profiles.

---

## Acceptance Criteria
- [ ] XTTS voice cloning supports multiple reference WAV files
- [ ] Recommended workflow supports six 8–10 second reference recordings
- [ ] Existing single-reference workflow remains supported where feasible
- [ ] Speaker reference discovery is centralized instead of duplicated across inference paths
- [ ] Speaker conditioning can be reused/cached when appropriate
- [ ] Reference audio validation provides clear errors
- [ ] Sample/config/API usage is updated
- [ ] README contains a concise voice cloning setup
- [ ] Detailed recording guide contains the six recommended sample texts
- [ ] Tests cover single-reference and multi-reference usage
- [ ] Documentation examples match the implemented API

---

## Technical Notes
- In `TTS.tts.models.xtts.Xtts.get_conditioning_latents(audio_path=...)`:
  - `speaker_embedding` is computed per clip and averaged via `torch.stack(speaker_embeddings).mean(dim=0)`. Thus, order does not alter speaker embedding values.
  - `gpt_cond_latents` concatenates audio (`torch.cat(audios, dim=-1)`) up to `gpt_cond_len=30` seconds. Because audio is concatenated in list order, **deterministic sorting** (e.g. `voice_01.wav` through `voice_06.wav`) is mandatory to guarantee exact repeatable conditioning and cache hits.
- XTTS internal resampling defaults to `load_sr = 22050 Hz`. High sample-rate recordings (44.1 kHz or 48 kHz 24-bit/16-bit PCM WAV) are smoothly resampled by XTTS's internal loader.

---

## Out of Scope
- XTTS-v2 model weight fine-tuning or custom checkpoint training
- Automatic in-browser microphone recording UI / wizard
- Real-time deep learning speech enhancement or noise reduction preprocessing
- Speaker diarization
- Changing XTTS tokenizer or vocoder architecture
- Alternative TTS backends (e.g. CosyVoice, F5-TTS)
